tests/test_header_matching.js

```diff
@@ -487,29 +487,34 @@ test('done does not fail when specified request header is not missing', t => {
   )
 })
 
-test('done fails when specified bad request header is present', t => {
-  nock('http://example.test', {
+test('when badheaders are present, badheaders prevents match', async t => {
+  const scope = nock('http://example.test', {
     badheaders: ['cookie'],
   })
-    .post('/resource')
+    .post('/')
     .reply(200, { status: 'ok' })
 
-  const d = domain.create()
+  await assertRejects(
```

Comment:
Now that `tap` supports asserting [rejected promises](https://node-tap.org/docs/api/asserts/#trejectspromise--fn-expectederror-message-extra), should we be using `t.rejects` instead of the separate `assert-rejects` lib?

Reply:
Sure, that sounds good. Would you like to take that on in a new PR?

`t.rejects` has been around for a while, though I had run into a bug which was causing us problems. The fix has been released, though: tapjs/node-tap#525.

Reply:
> Would you like to take that on in a new PR?

yeah, I can tackle that.

Reply:
On second thought I remember why I didn't switch back. We've talked about switching from tap to another test runner and if we do those calls will need to be updated again.

Given that, it doesn't seem worth updating all of those. Though if you've already started, it seems fine to finish. I don't feel strongly one way or another.